hive/steem/exceptions.py

```diff
@@ -0,0 +1,69 @@
+"""Defines exceptions which can be thrown by HttpClient."""
+
+def _str_trunc(value, max_length):
+    value = str(value)
+    if len(value) > max_length:
+        value = value[0:max_length] + '...'
+    return value
+
+class RPCError(Exception):
+    """Raised when an error is returned from upstream (jussi/steem)."""
+
+    @staticmethod
+    def build(error, body, index=None):
+        """Given an RPC error, builds exception w/ appropriate severity."""
+        assert 'message' in error, "missing error msg key: {}".format(error)
+        assert 'code' in error, "missing error code key: {}".format(error)
+
+        if isinstance(body, list):
+            item = body[index] if index else body[0]
+            method = item['method'] + ('[%s]' % index if index else '')
+            params = '[%s, (%d more)]' % (item['params'], len(body) - 1)
+        else:
+            method = body['method']
+            params = _str_trunc(body['params'], 1024)
+
+        message = RPCError.humanize(error)
+        message += ' in %s(%s)' % (method, params)
+
+        #if not RPCError.is_recoverable(error):
```

Comment:
since is_recoverable is stubbed out to return true for now, can this be un-commented?

Reply:
I suppose so. I was on the fence about removing this code altogether vs. moving to blacklist approach.